tests/end2end_test/__init__.py

```diff
@@ -0,0 +1,135 @@
+import time
+import logging
+from tests.end2end_test.end_to_end_test_helper import (
+    simple_query
+)
+from data_pipeline.data_store.s3_data_service import (
+    delete_s3_object
+)
+LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=too-few-public-methods, missing-class-docstring
+# pylint: disable=too-many-instance-attributes
+class DataPipelineCloudResource:
+    # pylint: disable=too-many-arguments
+    def __init__(
+            self, project_name, dataset_name, manuscript_table,
+            manuscript_version_table, person_table,
+            person_v2_table,
+            state_file_bucket_name, state_file_object_name
+    ):
+        self.project_name = project_name
+        self.dataset_name = dataset_name
+        self.manuscript_table = manuscript_table
+        self.manuscript_version_table = manuscript_version_table
+        self.person_table = person_table
+        self.person_v2_table = person_v2_table
+        self.state_file_bucket_name = state_file_bucket_name
+        self.state_file_object_name = state_file_object_name
+
+
+# pylint: disable=broad-except
+def truncate_table(
+        project_name: str,
+        dataset_name: str,
+        table_name: str
+):
+    try:
+        simple_query(
+            query=TestQueryTemplate.CLEAN_TABLE_QUERY,
+            project=project_name,
+            dataset=dataset_name,
+            table=table_name,
+        )
+    except Exception:
+        LOGGER.info("table not cleaned, maybe it does not exist")
+
+
+def delete_statefile_if_exist(
+        state_file_bucket_name,
+        state_file_object_name
+):
+
+    try:
+        delete_s3_object(state_file_bucket_name,
+                         state_file_object_name
+                         )
+    except Exception:
+        LOGGER.info("s3 object not deleted, may not exist")
+
+
+def get_table_row_count(
+        project_name,
+        dataset_name,
+        table_name
+):
+    query_response = simple_query(
+        query=TestQueryTemplate.READ_COUNT_TABLE_QUERY,
+        project=project_name,
+        dataset=dataset_name,
+        table=table_name,
+    )
+    return query_response[0].get("count")
+
+
+# pylint: disable=too-many-arguments
+def trigger_run_test_pipeline(
+        airflow_api, dag_id,
+        pipeline_cloud_resource: DataPipelineCloudResource
+):
+    for table in [
+            pipeline_cloud_resource.manuscript_table,
```

Comment:
Here you left out person_v2. It's not immediately clear why. I guess the test zip file doesn't contain any.
It seems in general this function just needs to know a list of table names to clear and check?

Reply:
the zip file i'm using doesn't contain person_v2, i will be changing the zip file lateron to the one with person_v2, the function needs to assert that some records have been written to some tables to clear. this can be parameterized with different with different zip files which (does not) contains person , and person_v2, and assert that records have been inserted (or not inserted) into one or the other table (depending).  This test will be a little bit more elaborate though

Reply:
The zip file will have `people` in the name. I guess we could just add two zip files to the test bucket?